components/automate-cli/cmd/chef-automate/ha_node_operations_utils_test.go

```diff
@@ -550,3 +550,111 @@ func TestGetIPsFromOSClusterResponseNotFound(t *testing.T) {
+			return nil
+		},
+		parseAndMoveConfigFilteToWorkspaceDirFunc: func(outFiles []string, outputDirectory string) error {
+			return nil
+		},
+	}
+
+	infra, _, err := mockUtil.getHaInfraDetails()
+	assert.NoError(t, err)
+
+	nodeUtil := NewNodeUtils(&MockRemoteCmdExecutor{
+		ExecuteFunc: func() (map[string][]*CmdResult, error) {
+			return nil, nil
+		},
+		ExecuteWithNodeMapFunc: func(nodeMap *NodeTypeAndCmd) (map[string][]*CmdResult, error) {
+			return nil, nil
+		},
+		GetSshUtilFunc: func() SSHUtil {
+			return &MockSSHUtilsImpl{
+				connectAndExecuteCommandOnRemoteFunc: func(remoteCommands string, spinner bool) (string, error) {
+					return "", nil
+				},
+				copyFileToRemoteFunc: func(srcFilePath string, destFileName string, removeFile bool) error {
+					return nil
+				},
+			}
+		},
+	}, command.NewMockExecutor(t), MockWriter.CliWriter)
+
+	t.Run("Get node object with to patch command in all nodes", func(t *testing.T) {
+
+		nodeObjects := getNodeObjectsToPatchWorkspaceConfigToAllNodes()
+		singleNode := true
+		outputDirectory := ""
+		err = executeCmdInAllNodeAndCaptureOutput(nodeObjects, singleNode, outputDirectory, infra, nodeUtil)
+		assert.Error(t, err, "No ips found")
+	})
+
+	t.Run("Get node object with to patch command for PG node", func(t *testing.T) {
+
+		nodeObjects := []*NodeObject{
+			NewNodeObjectWithOutputFile("", nil, nil, "", POSTGRESQL),
+		}
+		singleNode := true
+		outputDirectory := ""
+		err = executeCmdInAllNodeAndCaptureOutput(nodeObjects, singleNode, outputDirectory, infra, nodeUtil)
+		assert.Error(t, err, "No ips found")
+	})
+
+	t.Run("Get node object with to patch command for OS node", func(t *testing.T) {
+
+		nodeObjects := []*NodeObject{
+			NewNodeObjectWithOutputFile("", nil, nil, "", OPENSEARCH),
+		}
+		singleNode := true
+		outputDirectory := ""
+		err = executeCmdInAllNodeAndCaptureOutput(nodeObjects, singleNode, outputDirectory, infra, nodeUtil)
+		assert.Error(t, err, "No ips found")
+	})
+
+	t.Run("Get node object with to fetch config in all nodes", func(t *testing.T) {
+
+		nodeObjects := getNodeObjectsToFetchConfigFromAllNodeTypes()
+		singleNode := true
+		outputDirectory := ""
+		err = executeCmdInAllNodeAndCaptureOutput(nodeObjects, singleNode, outputDirectory, infra, nodeUtil)
+		assert.ErrorContains(t, err, "error on removing output header in fetched config")
+	})
+	// t.Run("No output file", func(t *testing.T) {
+
+	// 	nodeObjects := getNodeObjectsToFetchConfigFromAllNodeTypes()
+	// 	singleNode := true
+	// 	outputDirectory := ""
+	// 	for _, nodeObject := range nodeObjects {
+	// 		nodeObject.OutputFile = []string{}
+	// 	}
+	// 	err = nodeUtil.executeCmdInAllNodeAndCaptureOutput(nodeObjects, singleNode, outputDirectory, infra)
+	// 	assert.ErrorContains(t, err, "error on removing output header in fetched config")
+	// })
+
+}
+
+// func TestParseAndMoveConfigFilteToWorkspaceDir(t *testing.T) {
+// 	outDir := "/tmp/"
```

Comment:
if these test case and function are not needed can you remove them

Reply:
Done